### ISSUE TITLE
odd minimum in knockdown vs u

### ISSUE THREAD

**Issue by dankelley** [dankelley/mooring#10]
In a video (not published, but others in this repo have been told of it), I noticed that -- for a heavy line at least -- slowly increasing velo from 0 does *not* slowly increase knockdown.  Rather, the mooring gets *taller* for a while, and then gets knocked down.  I think this is because the drag bends the mooring over, which gives a force with a vertical component that can lift up more of the wire that was otherwise laying on the ground.  Well, that's what I came up with in the dentist's chair yesterday :-)

I made a test, as below.  Left: 1/4in line, which can be held up by the default float.  Right: 1/2in line (so, roughly 4X as mass per metre), which cannot be held up.  Black: case with delta-z of 0.1m; red: case with delta-z of 1m.

What I see is that delta-z doesn't matter much, but that that the odd behaviour only occurs if the mooring is too heavy for the float.

The code (in "develop" branch) for this is in `sandbox/dk/02_heavy_mooring` but I think it's interesting enough to put in a vignette.  (I don't, yet, have a proper vignette for the package.  The vignettes so far are mainly for developers.)

This is quite interesting, actually.  I think this might be how spiders (or some other bugs, not sure) can spit out a line into the air, and the air catches it and pulls them off the ground.  The idea would be similar: current bends the line, now the drag has a vertical component that can possibly outweigh the gravity component.  What do you think of this, @richardsc and @clayton33?  (And is Jay on GH?  It would be good to be able to use the at syntax so Jay can see such things and give ideas.)

<details>

---
author: kelley
date: 2021-06-02
output: "reprex::reprex\\_document"
title: soot-cub\_reprex.R
---

``` r
library(mooring)
a <- anchor(depth=1000) # make it deep so a heavy wire can't be held up
par(mfrow=c(1, 2), mar=c(3,3,1,1), mgp=c(2,0.7,0))
wiretypes <- c("1/4in wire/jack", "1/2in wire/jack")
for (wiretype in wiretypes) {
    w <- wire(wiretype, length=900)
    f <- float()
    m <- mooring(a, w, f)
    us <- seq(0, 1.5, 0.05)
    first <- TRUE
    for (dz in c(0.1, 1)) {
        md <- discretise(m, dz)
        zs <- sapply(us, function(u)
                     {
                         mdk <- knockdown(md, u)
                         z(tail(mdk,1))
                     }
        )
        if (first) {
            plot(us, zs, type="o", pch=20, cex=0.5, xlab="Horiz. velo. [m/s]", ylab="Float z [m]")
            first <- FALSE
        } else {
            lines(us, zs, type="o", pch=20, cex=0.5, col=2)
        }
    }
    legend("topright", lwd=1, col=c(1,2), legend=c(0.1, 1), title="dz")
    mtext(paste0("wire type: ", wiretype))
}
```

![](https://i.imgur.com/Gao3yk7.png)

<sup>Created on 2021-06-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by richardsc** [comment]
Oh, for a Tim's coffee and a meeting-less afternoon standing in your office at the blackboard ...

**Comment by dankelley** [comment]
Tagging @FormulaFunRacing, for his info.

**Comment by clayton33** [comment]
Tell me about it @richardsc , I miss those blackboard afternoons so much.

**Comment by dankelley** [comment]
Me too.  Or me three, I guess.  Sigh.

**Comment by dankelley** [comment]
This was an artifact of the mooring being "buried" in the sediments (see #12).  Now, if buoyancy cannot support a portion of the mooring, then that portion is laid out on the ground, in the positive x direction. 

With the fix for #12 ("develop" branch, commit 6027a9c69381c563d58ed35f6e68ebad196b1c8a), we also get a fix. for this as well (see Details).

Now, currents knock down moorings, as expected.  No more need for mental gymnastics!



<details>

``` r
library(mooring)
a <- anchor(depth=1000) # make it deep so a heavy wire can't be held up
par(mfrow=c(1, 2), mar=c(3,3,1,1), mgp=c(2,0.7,0))
wiretypes <- c("1/4in wire/jack", "1/2in wire/jack")
for (wiretype in wiretypes) {
    w <- wire(wiretype, length=900)
    f <- float()
    m <- mooring(a, w, f)
    us <- seq(0, 1.5, 0.05)
    first <- TRUE
    for (dz in c(0.1, 1)) {
        md <- discretise(m, dz)
        zs <- sapply(us, function(u)
                     {
                         mdk <- knockdown(md, u)
                         z(tail(mdk,1))
                     }
        )
        if (first) {
            plot(us, zs, type="o", pch=20, cex=0.5, xlab="Horiz. velo. [m/s]", ylab="Float z [m]")
            first <- FALSE
        } else {
            lines(us, zs, type="o", pch=20, cex=0.5, col=2)
        }
    }
    legend("topright", lwd=1, col=c(1,2), legend=c(0.1, 1), title="dz")
    mtext(paste0("wire type: ", wiretype))
}
```

![](https://i.imgur.com/gLzMt1u.png)

<sup>Created on 2021-06-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
